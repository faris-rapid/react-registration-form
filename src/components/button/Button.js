import styled, { css } from 'styled-components';

const Button = styled.button`
	background: ${(props) => (props.color ? props.color : 'tomato')};
	border-radius: 5px;
	border: 2px solid;
	color: white;
	margin: 1rem 1em;
	padding: 0.8em 1.2em;
	cursor: pointer;

	${(props) =>
		props.primary &&
		css`
			color: white;
		`}
	${(props) =>
		props.outline &&
		css`
			background: white;
			color: ${(props) => (props.color ? props.color : 'tomato')};
			border: 2px solid ${(props) => (props.color ? props.color : 'tomato')};
		`}
		${(props) =>
		props.pill &&
		css`
			border-radius: 20px;
		`}
		${(props) =>
		props.square &&
		css`
			border-radius: 0px;
		`}
		${(props) =>
		props.disabled &&
		css`
			opacity: 0.65;
			cursor: not-allowed;
		`};
`;

export default Button;
