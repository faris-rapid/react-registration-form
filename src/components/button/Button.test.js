import React from 'react';
import * as renderer from 'react-test-renderer';
import Button from './Button';

describe('Button tests', () => {
	it('snapshot testing Button componen', () => {
		const wrapper = renderer.create(<Button />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
