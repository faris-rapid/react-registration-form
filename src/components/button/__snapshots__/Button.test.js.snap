// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests snapshot testing Button componen 1`] = `
<button
  className="sc-bdnxRM gAkMzT"
/>
`;
